examples/protonect/Protonect.cpp

```diff
@@ -87,6 +91,10 @@ int main(int argc, char *argv[])
     cv::imshow("ir", cv::Mat(ir->height, ir->width, CV_32FC1, ir->data) / 20000.0f);
     cv::imshow("depth", cv::Mat(depth->height, depth->width, CV_32FC1, depth->data) / 4500.0f);
 
+    if (!registered) registered = new unsigned char[depth->height*depth->width*rgb->bytes_per_pixel];
+    registration->apply(rgb,depth,registered);
+    cv::imshow("registered", cv::Mat(depth->height, depth->width, CV_8UC3, registered));
```

Comment:
Now that I think about it, it would probably be better to switch to 4 BPP anyway and use the extra channel to add depth data. That would make filtering a lot easier, and would probably also be faster than shifting around single bytes.